File: test/c++11/intel_x64/guest_interrupt_status.cpp
#include <pal/vmcs/guest_interrupt_status.h>
#include <stdio.h>

void test_guest_interrupt_status_compile(void)
{
    // Register accessors
    pal::guest_interrupt_status::set(0xA55A);
    auto value = pal::guest_interrupt_status::get();

    // Field accessors
    pal::guest_interrupt_status::rvi::get();
    pal::guest_interrupt_status::rvi::get(value);
    pal::guest_interrupt_status::rvi::set(0x0);
    pal::guest_interrupt_status::rvi::set(0x0, value);

    // Printers
    pal::guest_interrupt_status::print(printf);
    pal::guest_interrupt_status::print(printf, value);
    pal::guest_interrupt_status::rvi::print(printf);
    pal::guest_interrupt_status::rvi::print(printf, value);
}

File: test/rust/intel/src/guest_interrupt_status.rs
use pal;

pub fn test_guest_interrupt_status_compile()
{
    // Register accessors
    pal::vmcs::guest_interrupt_status::set(0xA55A);
    let mut value = pal::vmcs::guest_interrupt_status::get();

    // Field accessors
    pal::vmcs::guest_interrupt_status::get_rvi();
    pal::vmcs::guest_interrupt_status::get_rvi_from_value(value);
    pal::vmcs::guest_interrupt_status::set_rvi(0x0);
    pal::vmcs::guest_interrupt_status::set_rvi_in_value(0x0, &mut value);

    // Printers
    pal::vmcs::guest_interrupt_status::print();
    pal::vmcs::guest_interrupt_status::print_from_value(value);
    pal::vmcs::guest_interrupt_status::print_rvi();
    pal::vmcs::guest_interrupt_status::print_rvi_from_value(value);
}
